src/main/java/com/boomi/flow/services/boomi/mdh/match/MatchResponseMapper.java

```diff
@@ -0,0 +1,99 @@
+package com.boomi.flow.services.boomi.mdh.match;
+
+import com.boomi.flow.services.boomi.mdh.records.GoldenRecordConstants;
+import com.boomi.flow.services.boomi.mdh.universes.Universe;
+import com.manywho.sdk.api.run.elements.type.MObject;
+import com.manywho.sdk.api.run.elements.type.Property;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class MatchResponseMapper {
+
+    public static List<MObject> createMobjectsFromResults(Universe universe, String universeId, List<MatchEntityResponse.MatchResult> results) {
+        return results.stream()
+                .filter(result -> result.getEntity() != null && result.getEntity().get(universe.getName()) != null)
+                .map(result -> MatchResponseMapper.createMobjectFromResult(universeId, universe, result))
+                .collect(Collectors.toList());
+
+    }
+
+    private static MObject createMobjectFromResult(String universeId, Universe universe, MatchEntityResponse.MatchResult result) {
+        var externalId = result.getEntity().get(universe.getName()).get(universe.getIdField()).toString();
+
+        var propertiesMatched = new Property(FuzzyMatchDetialsConstants.MATCH, new ArrayList<>());
+        var propertiesDuplicated = new Property(FuzzyMatchDetialsConstants.DUPLICATE, new ArrayList<>());
+        var propertiesAlreadyLinked = new Property(FuzzyMatchDetialsConstants.ALREADY_LINKED, new ArrayList<>());
+        var object = new MObject(universeId + " match", externalId, Arrays.asList(propertiesMatched, propertiesDuplicated, propertiesAlreadyLinked));
+        object.setTypeElementBindingDeveloperName(object.getDeveloperName());
+
+        var properties =  new ArrayList<Property>();
+        result.getEntity().get(universe.getName())
+                .forEach((key, value) -> properties.add(new Property(key, (String) value)));
+
+        properties.add(new Property(GoldenRecordConstants.SOURCE_ID_FIELD, result.getIdResource()));
+        properties.add(new Property(FuzzyMatchDetialsConstants.FUZZY_MATCH_DETAILS, (MObject) null));
+        properties.addAll(object.getProperties());
+
+        object.setProperties(properties);
+
+        if ("SUCCESS".equals(result.getStatus())) {
+            result.getMatch().forEach(match -> addMatchesToProperty(propertiesMatched, universe, match, universe.getIdField(), true));
+            result.getDuplicate().forEach(match -> addMatchesToProperty(propertiesDuplicated, universe, match, universe.getIdField(), true));
+        } else if ("ALREADY_LINKED".equals(result.getStatus())) {
+            var entityH = result.getEntity().get(universe.getName());
+            addAlreadyLinked(propertiesAlreadyLinked, universe, entityH, universe.getIdField());
+        }
+
+        return object;
+    }
+
+
+    private static void addAlreadyLinked(Property alreadyLinkedProperty, Universe universe, Map<String, Object> entity, String idField){
```

Comment:
The mutation in this method seems very smelly too - I think this could be a `map` function too (i.e. pure, and returning rather than mutating)